src/phoenix/server/api/types/Span.py

```diff
@@ -330,3 +333,16 @@ def _hide_embedding_vectors(
         _embeddings.append(_embedding)
     _attributes[EMBEDDING_EMBEDDINGS] = _embeddings
     return _attributes
+
+
+def _convert_metadata_to_string(metadata: Any) -> Optional[str]:
+    """
+    Converts metadata to a string representation.
+    """
+
+    if metadata is None or isinstance(metadata, str):
+        return metadata
+    try:
+        return json.dumps(metadata)
+    except json.JSONDecodeError:
```

Comment:
the most likely error is actually `TypeError: Object of type object is not JSON serializable`
```suggestion
    except Exception:
```